src/main/tut/2.2-limits-and-colimits.md

```diff
@@ -0,0 +1,275 @@
+```Haskell
+p' = p . m
+q' = q . m
+```
+```scala
+val p' = p compose m
+val q' = q compose m
+
+```
+...................
+
+```Haskell
+f :: c' -> c
+```
+```scala
+val f : C' => C
+```
+..................
+```Haskell
+u :: c -> Lim D
+```
+```scala
+val u :  C => Lim[D]
+```
+..................
+```Haskell
+u . f :: c' -> Lim D
+```
+```scala
+u compose f : C' => Lim[D]
+```
+..................
+```Haskell
+contramap :: (c' -> c) -> (c -> Lim D) -> (c' -> Lim D)
+contramap f u = u . f
+```
+```scala
+def contramap[C, C', D[_]](f: C' => C, u: C => Lim[D]): (C' => Lim[D]) = 
+  u compose f
+```
+..................
+```Haskell
+f :: c' -> c
+```
+```scala
+val f : C' => C
+```
+..................
+```Haskell
+Nat(Δc, D) -> Nat(Δc', D)
+```
+```scala
+(Δc ~> D) => (Δc' ~> D)
+```
+..................
+```Haskell
+αa :: Δca -> D a
+```
+```scala
+val αA : ΔcA => D[A]
+```
+..................
+```Haskell
+αa :: c -> D a
+```
+```scala
+val αA : C => D[A]
+```
+..................
+```Haskell
+βA :: c' -> D A
+```
+```scala
+val βA : C' => D[A]
+```
+..................
+```Haskell
+βa = αa . f
+```
+```scala
+val βA = αA compose f 
+```
+..................
+```Haskell
+c -> Nat(Δc, D)
+```
+```scala
+C => (Δc ~> D) 
+```
+..................
+```Haskell
+f :: a -> b
+g :: a -> b
+```
+```scala
+val f : A => B
+val g : A => B 
+```
+..................
+```Haskell
+p :: c -> a
+q :: c -> b
+```
+```scala
+val p : C => A
+val q : C => B
+```
+..................
+```Haskell
+q = f . p
+q = g . p
+```
+```scala
+q == f compose p
+q == g compose p
+```
+
+
+..................
+
+inline code below
+```Haskell
+q = f . p
+```
+```scala
+q = f compose p
+```
+
+
+..................
+```Haskell
+f . p = g . p
+```
+```scala
+f compose p == g compose p
+```
+
+
+..................
+```Haskell
+f (x, y) = 2 * y + x
+g (x, y) = y - x
+```
+```scala
+def f(x, y) = 2 * y + x
+def g(x, y) = y - x
+```
+
+..................
+```Haskell
+p t = (t, (-2) * t)
+```
+```scala
+def p(t) = (t, (-2) * t)
+```
+
+..................
+```Haskell
+f . p' = g . p'
+```
+```scala
+f compose p' == g compose p'
+```
+
+..................
+```Haskell
+p'() = (0, 0)
+```
+```scala
+def p'[U](u: U) = u match {
+  case _: Unit => (0, 0) 
+} 
```

Comment:
Not too sure about this part. Is there a better way?

Reply:
Right, the fact that  `Unit`, the empty parameter list and the singleton Tuple are 3 different things in scala, is very annoying. The translation above is incorrect however I think: partial, and non parametric.

How about:
```scala
def p: Unit => (Int, Int) = _ => (0,0)
```